back/app/services/idea_custom_fields_service.rb

```diff
@@ -137,21 +137,6 @@ def remove_ignored_update_params(field_params)
     field_params.except(:code, :input_type)
   end
 
-  def check_form_structure(fields_from_params, errors)
-    return if fields_from_params.empty?
-
-    unless fields_from_params.first[:input_type] == 'page'
-      error = { error: "First field must be of type 'page'" }
-      errors['0'] = { structure: [error] }
-    end
-
-    # Check the last field is a page
-    last_field = fields_from_params.last
-    unless last_field[:input_type] == 'page' && last_field[:key] == 'form_end'
-      errors[(fields_from_params.length - 1).to_s] = { structure: [{ error: "Last field must be of type 'page' with a key of 'form_end'" }] }
-    end
-  end
-
```

Comment:
I see you made the opposite decision to me - ie moved this into the controller rather than out of it! Still think eventually we'll move it out into a new service if we refactor update_all

Reply:
Sorry I didn't realize that 😅 This part was very confusing in the diffs, but yes sounds good to move it out of the controller at some point.